src/app/parcels/edit/EditParcelForm.tsx

```diff
@@ -111,19 +115,18 @@ const EditParcelForm = ({ parcelId }: EditParcelFormProps): React.ReactElement =
             );
             setPackingSlotsIsShown(parcelData.packing_slot?.is_shown);
             setCollectionCentreIsShown(parcelData.collection_centre?.is_shown === true);
-
             setIsLoading(false);
         })();
     }, [parcelId]);
 
     const initialFormErrors: ParcelErrors = {
         voucherNumber: Errors.none,
         packingDate: Errors.none,
-        packingSlot: Errors.none,
+        packingSlot: packingSlotIsShown ? Errors.none : Errors.invalidPackingSlot,
```

Comment:
I have moved the logic in showing the initial errors into EditParcelForm from ParcelForm.tsx. This addition should be equivalent to the use states that I've removed, with fewer lines. Lmk if you disagree. The two useStates that I've removed only activate when editMode is true (which is only in editParcelForm), so make sense to just be in EditParcelForm? Also am not sure why they would need to be in a useState? Have tested that when you turn off the packing slot of a parcel in the admin controls, when you click on editParcel, the error message is initially shown.

Reply:
This is great, nice job!

I think there is the possibility of a user being on the edit parcel page, choosing a valid slot, and another user then turning off that packing slot in the admin panel, then the first user submitting the parcel without error. However, I think this currently possible anyway as I don't think `packingSlotIsShown` is refetched when the database changes (although I haven't checked). So I think this change is okay.

Reply:
Aah good point, though yeah I don't think it would have updated correctly in its current state. We could always edit this if we do need to change.